src/components/transport_manager/src/transport_manager_impl.cc

```diff
@@ -540,14 +540,19 @@ void TransportManagerImpl::AddConnection(const ConnectionInternal& c) {
   connections_.push_back(c);
 }
 
-void TransportManagerImpl::RemoveConnection(uint32_t id) {
+void TransportManagerImpl::RemoveConnection(
+    uint32_t id, transport_adapter::TransportAdapter* transport_adapter) {
```

Comment:
@AByzhynar thank you for your comment. In this case I think the pointer is more suitable. This is a private method so calls to it are limited inside the class, where a pointer to TransportAdapter is used. However there is no guarantee that this pointer is not NULL. So before dereferencing it should be checked. If a reference is used in signature, this should be done each time the method is called - now twice. If a pointer is used, then check can be done only inside the method as it is done now. I think this is the proper design for this case and I am aware that pointer usage in C++ code should be reduced.